src/foundry/foundry.js/hooks.d.ts

```diff
@@ -1,830 +1,804 @@
      * @remarks This is called by {@link Hooks.call}.
-     * @see {@link ActorSheet#_onDrop}
+     * @see {@link Application#close}
      */
-    dropActorSheetData: (actor: Actor, sheet: ActorSheet, data: ActorSheet.DropData.Combined) => unknown;
+    type CloseApplication<A extends Application = Application> = (app: A, html: JQuery) => boolean | void;
 
     /**
-     * This is called during the drop portion of a drag-and-drop event on a canvas.
-     * @param canvas - the Canvas the data has been dropped on
-     * @param data   - the dropped data, already parsed as an object via JSON
-     * @remarks This is called by {@link Hooks.call}.
-     * @see {@link Canvas#_onDrop}
+     * @remarks This is called after assuming or releasing control over a {@link PlaceableObject}.
+     * @param object     - the PlaceableObject
+     * @param controlled - whether the PlaceableObject is controlled
+     * @typeParam P      - the type of the PlaceableObject
+     * @remarks The name for this hook is dynamically created by joining 'control' and the type name of the
+     * PlaceableObject.
+     * @remarks This is called by {@link Hooks.callAll}.
+     * @see {@link PlaceableObject#control}
+     * @see {@link PlaceableObject#release}
      */
-    dropCanvasData: (
-      canvas: Canvas,
-      data: TokenLayer.DropData | NotesLayer.DropData | User.DropData | TilesLayer.DropData
+    type ControlPlaceableObject<P extends PlaceableObject = PlaceableObject> = (
+      object: P,
+      controlled: boolean
     ) => unknown;
 
     /**
-     * This is called during the drop portion of a drag-and-drop event on a roll table.
-     * @param entity - the Entity the table belongs to
-     * @param config - the RollTableConfig
-     * @param data   - the dropped data, already parsed as an object via JSON
-     * @remarks This is called by {@link Hooks.call}.
-     * @see {@link RollTableConfig#_onDrop}
+     * A hook event that fires for every embedded Document type after conclusion of a creation workflow.
+     * Substitute the Document name in the hook event to target a specific type, for example "createToken".
+     * This hook fires for all connected clients after the creation has been processed.
+     *
+     * @param document - The new Document instance which has been created
+     * @param options  - Additional options which modified the creation request
+     * @param userId   - The ID of the User who triggered the creation workflow
+     * @remarks The name for this hook is dynamically created by joining 'create' and the type name of the Document.
+     * @remarks This is called by {@link Hooks.callAll}.
+     * @see {@link ClientDatabaseBackend#_postCreateDocumentCallbacks}
+     * @see {@link TokenDocument#_onUpdateTokenActor}
      */
-    dropRollTableSheetData: (entity: Entity, config: RollTableConfig, data: object) => unknown;
+    type CreateDocument<D extends DocumentConstructor = DocumentConstructor> = (
```

Comment:
The documentation in foundry (and here) is a bit incorrect this is not only about embedded documents but about _all_ documents. Consider creating an issue.